packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php

```diff
@@ -20,16 +20,15 @@ public static function select(
         Scope $scope
     ): ParametersAcceptor {
         $variants = $reflection->getVariants();
+        
         if ($node instanceof FunctionLike) {
             return ParametersAcceptorSelector::selectSingle($variants);
         }
-
+        
         if ($node->isFirstClassCallable()) {
             return ParametersAcceptorSelector::selectSingle($variants);
         }
 
-        return count($variants) > 1
-            ? ParametersAcceptorSelector::selectFromArgs($scope, $node->getArgs(), $variants)
-            : ParametersAcceptorSelector::selectSingle($variants);
+        return ParametersAcceptorSelector::selectFromArgs($scope, $node->getArgs(), $variants);
```

Comment:
`selectFromArgs()` is slower than `selectSingle()` as far as I remember, as it check many things, verify count() is only 1 was already faster.

Reply:
at least in phpstan itself we try to get rid of `selectSingle` occurences and use `selectFromArgs` where possible.

I did not yet see it in perf profiles, but you are likely right, that `selectSingle` is marginally faster

Reply:
`count()` without recursive call is cheap, while call `selectFromArgs()` that check many things already will be slower, so I prefer the original implementation to check `count()` first, then use `selectSingle()` when possible, fallback to `selectFromArgs()` on multi variants.

Reply:
👍